lib/ecto/adapters/myxql/connection.ex

```diff
@@ -312,16 +312,27 @@ if Code.ensure_loaded?(MyXQL) do
       error!(query, "MySQL adapter does not support materialized CTEs")
     end
 
-    defp cte_expr({name, _opts, cte}, sources, query) do
-      [quote_name(name), " AS ", cte_query(cte, sources, query)]
+    defp cte_expr({name, opts, cte}, sources, query) do
+      operation = case opts[:operation] do
+        :update_all -> :update_all
+        :delete_all -> :delete_all
+        :insert_all -> :insert_all
+        _ -> :all
+      end
```

Comment:
As per my comment on Ecto, we should guarantee the `.operation` field always exist and it is never nil. Once we tidy up the Ecto side, we can revisit this one. :)

Reply:
I've just updated the code to ensure `.operation` and handle `nil`, now that Ecto PR has been merged